ament_cmake_python/ament_cmake_python-extras.cmake

```diff
@@ -46,7 +46,7 @@ macro(_ament_cmake_python_get_python_install_dir)
     set(_python_code
       "import os"
       "import sysconfig"
-      "print(os.path.relpath(sysconfig.get_path('purelib', vars={'base': '${CMAKE_INSTALL_PREFIX}'}), start='${CMAKE_INSTALL_PREFIX}').replace(os.sep, '/'))"
+      "print(os.path.relpath(sysconfig.get_path('purelib', vars={'base': '${CMAKE_INSTALL_PREFIX}'}, scheme='posix_prefix'), start='${CMAKE_INSTALL_PREFIX}').replace(os.sep, '/'))"
```

Comment:
Just a question; is "posix_prefix" the correct thing to do on Windows?

Reply:
Maybe not. It does change the paths away from what distutils used to report.

```python3
>>> import sysconfig
>>> sysconfig.get_path('purelib', vars={'base': '.'})
'Lib\\site-packages'
>>> sysconfig.get_path('purelib', vars={'base': '.'}, scheme='posix_prefix')
'lib\\python3.10\\site-packages'
>>> from distutils.sysconfig import get_python_lib
>>> get_python_lib(prefix='.')
'.\\Lib\\site-packages'
```

Reply:
Maybe we could mirror the same logic as cpython to conditionally enforce the scheme using `os.name`:
https://github.com/python/cpython/blob/7cdaf87ec50f76c934ba651256484c4624b84ef2/Lib/sysconfig.py#L278-L295

Reply:
Ooh - even better. `scheme='posix_prefix' if sysconfig.get_default_scheme() == 'posix_local' else None`.

That should only enforce the scheme if the debian hack is present.

Reply:
Drat - `None` doesn't work for `scheme`.
Maybe something like:
`get_path_kwargs = {'scheme': 'posix_prefix'} if sysconfig.get_default_scheme() == 'posix_local' else {}`
and then pass `**get_path_kwargs` to `get_path`?

Reply:
Or maybe:
`scheme={'posix_local': 'posix_prefix'}.get(sysconfig.get_default_scheme(), sysconfig.get_default_scheme())`
or
`scheme='posix_prefix' if sysconfig.get_default_scheme() == 'posix_local' else sysconfig.get_default_scheme()`